Makefile

```diff
@@ -19,14 +21,32 @@ export MONO_OPTIONS
 endif
 
 include build-tools/scripts/msbuild.mk
-
 all::
 	$(MSBUILD) $(MSBUILD_FLAGS) $(SOLUTION)
 
 all-tests::
 	MSBUILD="$(MSBUILD)" tools/scripts/xabuild $(MSBUILD_FLAGS) Xamarin.Android-Tests.sln
-
-prepare:: prepare-msbuild
+	
+prepare:: linux-prepare prepare-msbuild
```

Comment:
I'm still not sure why you appear to dislike:

```make
ifeq ($(OS),Linux)
prepare:: linux-prepare
endif # $(OS)=Linux
```

This would avoid the need for the `if [ "$(OS)" = "Linux" ]` check in `linux-prepare`, as `linux-prepare` wouldn't be called on non-Linux platforms.